src/main/java/com/iexec/common/contract/IexecSmartContractConnectionPolicy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.contract;
+
+import com.iexec.common.contract.generated.IexecHubContract;
+import lombok.extern.slf4j.Slf4j;
+
+import java.math.BigInteger;
+
+@Slf4j
+public class IexecSmartContractConnectionPolicy {
```

Comment:
Isn't it more of a `IexecHubSmartContractValidator/Checker`?
Also `expectedFinalDeadlineRatio` is a constant variable that depends on the contract, so I think it can be encapsulated in the class and there is no need to pass it as an argument.

Reply:
`Validator` would be great as it would look like [`TeeEnclaveConfigurationValidator` class](https://github.com/iExecBlockchainComputing/iexec-common/blob/develop/src/main/java/com/iexec/common/tee/TeeEnclaveConfigurationValidator.java).

I'd like to encapsulate `expectedFinalDeadlineRatio` in the class, but it looks like there's currently no value injection in `iexec-common`. I was unsure whether it was intentional so I didn't want to bother with it. Do you think it'd be great to add this feature to Common?

Reply:
I mean adding a constant `EXPECTED_FINAL_DEADLINE_RATIO` in the class `IexecSmartContractValidator`.

Reply:
@jeremyjams suggested that this value should be parametrized so that we could update it if evolves one day. I agree passing it through the whole chain is not really clean but I haven't find any better solution :(

Reply:
@jeremyjams If the value evolves, that means the smart contract also has evolved and we need to update iexec-common in all cases. So I find it coherent to use the constant.

Reply:
The other option would be to not have an `expectedValue` and use the value fetched from the smart contract. We only need to make sure it is `> 0`.

Reply:
I agree, what truly matters is `final_deadline_ratio()` call returns something greater than zero (meaning the remote function exists).

Reply:
I think I agree with @zguesmi. That probably means we don't need those value injections... And that simplifies the whole logic of the thing.

Reply:
Alright, so let's remove this whole `expectedValue` thing! :)